src/app/App.tsx

```diff
@@ -443,6 +471,42 @@ class App extends AsyncComponent<RouteComponentProps, AppState> {
 
         document.title = title;
     }
+
+    /**
+     * Refresh the token one minute before it expires.
+     */
+    private validateTokenPeriodically() {
+        if (this._tokenExpiryTimer !== undefined) {
+            clearInterval(this._tokenExpiryTimer);
+            this._tokenExpiryTimer = undefined;
+        }
+        const jwt = this._storageService.load<string>("dashboard-jwt");
+        const expiryTime = this.getTokenExpiry(jwt);
+
+        this._tokenExpiryTimer = setInterval(async () => {
+            if (expiryTime < moment().valueOf()) {
+                this._authService.logout();
+                clearInterval(this._tokenExpiryTimer);
+                this._tokenExpiryTimer = undefined;
+            } else if (expiryTime < (moment().valueOf() + MilliSecondsInOneMinute)) {
```

Comment:
Let's leverage the `moment` API since we are already using it, to make this much more nice and readable.